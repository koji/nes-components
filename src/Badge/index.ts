export * from './Badge';
export { default as NesBadge } from './Badge';
