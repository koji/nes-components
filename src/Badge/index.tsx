import * as React from "react";
import "../../node_modules/nes.css/css/nes.css";

type Props = {
  badgeType: "dark" | "primary" | "success" | "warning" | "error" ;
  isSplited: boolean;
  splits?: Array<"dark" | "primary" | "success" | "warning" | "error">;
  // badgeSize: "large" | "medium" | "small";
  contents: string[];
};

export class NesBadge extends React.Component<Props> {
  render() {
    const { badgeType, isSplited, splits, /* badgeSize, */ contents } = this.props;
    const badgeClass = `is-${badgeType}`;
    return (
      <React.Fragment>
      {isSplited ? (
        splits && (
          <a href="#" className="nes-badge is-splited">
            <span className={`is-${splits[0]}`}>{contents[0]}</span>
            <span className={`is-${splits[1]}`}>{contents[1]}</span>
          </a>
        )
        ):(
        <a href="#" className="nes-badge">
          <span className={badgeClass}>{contents[0]}</span>
        </a>
      )}
      </React.Fragment>
    );
  }
}
