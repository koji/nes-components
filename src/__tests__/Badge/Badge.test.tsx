import React from 'react';
// import { render, cleanup } from '@testing-library/react';
import { NesBadge } from '../../Badge';
// import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';

describe('<NesBadge>', () => {
  // it('renders without crashing', () => {
  //   const { getAllByDisplayValue } = render(
  //     <NesBadge badgeType={['primary']} contents={['nes-components']} />
  //   );
  //   expect(getAllByDisplayValue).toBe('nes-components');
  // });

  it('renders without crashing', () => {
    const tree = renderer
      .create(
        <NesBadge badgeType={['primary']} contents={['nes-components']} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
