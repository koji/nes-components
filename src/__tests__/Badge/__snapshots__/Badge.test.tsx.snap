// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NesBadge> renders without crashing 1`] = `
<a
  className="nes-badge"
  href="#"
>
  <span
    className="is-primary"
  >
    nes-components
  </span>
</a>
`;
